src/Core/Usecase/Message/ReceiveMessage.php

```diff
@@ -107,120 +107,139 @@ public function setTargetedSurveyStateRepo(Entity\TargetedSurveyStateRepository
 	 */
 	private function createOutgoingMessage($contact_id, $survey_state_entity, $next_form_attribute)
 	{
-		// create message that we will send to thhe user next
-		$newMessage = $this->repo->getEntity();
-		$messageState = array(
+		// Create new message to send next question to the user
+		$outgoingMessage = $this->repo->getEntity()->setState([
 			'contact_id' => $contact_id,
 			'post_id' => $survey_state_entity->post_id,
 			'title' => $next_form_attribute->label,
 			'message' => $next_form_attribute->label,
 			'status' => Message::PENDING,
-			'type' => 'sms',//FIXME
+			'type' => 'sms', // FIXME
 			'direction' => Message::OUTGOING
-		);
-		$newMessage->setState($messageState);
-		$this->outgoingMessageValidator->check($messageState);
-		$newMessageId = $this->repo->create($newMessage);
-		if (!$newMessageId) {
-			\Log::error(
-				'Could not create new incoming message',
-				compact('contact_id')
-			);
-			throw new HTTP_Exception_400('Could not create new outgoing message for contact_id: ' . $contact_id);
+		]);
+
+		// Verify its valid
+		// @todo not sure we even need to bother. If its not valid, all is lost.
+		if (!$this->outgoingMessageValidator->check($outgoingMessage->asArray())) {
+			$this->validatorError($outgoingMessage);
 		}
-		return $newMessageId;
+
+		// Save the message
+		$outgoingMessageId = $this->repo->create($outgoingMessage);
+
+		// But then continue anyway
+		return $outgoingMessageId;
 	}
 
 	/**
-	 * @param $contact_id
 	 * @param $incoming_message
-	 * @throws HTTP_Exception_400
 	 */
-	private function createTargetedSurveyMessages($contact_id, $incoming_message)
+	private function createTargetedSurveyMessages($incoming_message)
 	{
-		$surveyStateEntity = $this->targeted_survey_state_repo->getByContactId($contact_id);
-		$messageInSurveyState = clone $this->repo;
-		// ... attempt to load the entity
-		$messageInSurveyState = $messageInSurveyState->get($surveyStateEntity->message_id);
+		// Load the survey state for this contact
+		$surveyStateEntity = $this->targeted_survey_state_repo->getByContactId($incoming_message->contact_id);
+
+		// Attempt to load the previous message
+		$messageInSurveyState = $this->repo->get($surveyStateEntity->message_id);
+
+		// If we didn't find an outgoing message
 		if (!$messageInSurveyState || $messageInSurveyState->direction !== \Ushahidi\Core\Entity\Message::OUTGOING) {
-			//we can't save it as a message of the survey
+			// We can't save it as a message of the survey
+			// ... log an error because we should probably never end up here
 			\Log::error(
 				'Could not add contact\'s  message',
 				[
-					'contact_id' => $contact_id,
+					'contact_id' => $incoming_message->contact_id,
 					'form_id' => $surveyStateEntity->form_id
 				]
 			);
-			throw new HTTP_Exception_400(
-				'Outgoing question not found for contact ' . $contact_id . ' and form ' . $surveyStateEntity->form_id
-			);
+
+			// @todo Should we actually create a new post if this happens?!
```

Comment:
YES. If this happens and we don't reject, at least we should make sure it doesn't end up in a targeted survey's post.

Reply:
So that already happens. It won't end up in the targetted survey *but* it ends up nowhere. It doesn't get its own post, and it doesn't go on the targetted survey.

I should probably refactor a bit more to make match the non-targeted survey flow